File: assessment 3/linkedlist.java
import java.util.Scanner;
import java.util.LinkedList;

public class linkedlist {
    public static void main(String[] args) {
        linkedlist<String> linkedlist = new linkedlist<>();
        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the 4 elements for the Linkedlist");
        for(int i=0;i<4;i++){
            System.out.println("Element "+(i+1)+":");
            String element = sc.nextLine();
            linkedlist.add(element);
        }

        System.out.println("Original LinkedList: "+linkedlist);
        System.out.println("Enter the inddex of the element to be removed");
        int index = sc.nextInt();

        if(index>=0 && index<4){
            linkedlist.remove(index);
            System.out.println("Linkedlist after removing the index: "+linkedlist);}
        else{
            System.out.println("Invalid Index");
        }
        scanner.close();
    }
}

File: assessment 3/linked_list.py
class Node:
    def __init__(self,data):
        self.data=data
        self.next=None

class LinkedList:
    def  __init__  (self):
        self.head=None

    def append(self,data):
        new_node=Node(data)
        if not self.head:
            self.head=new_node
            return
        current=self.head
        while current.next:
            current=current.next
        current.next=new_node

    def remove_at_index(self, index):
        if self.head is None:
            return "List is empty"

        if index==0:
            self.head=self.head.next
            return

        current=self.head
        for _ in range(index-1):
            if current is None or current.next is None:
                return "Index out of bounds"
            current=current.next

        if current.next is None:
            return"Index out of bounds"

        current.next=current.next.next

    def print_list(self):
        current=self.head
        while current:
            print(current.data, end="->")
            current=current.next
        print("None") 

    def create_list():
        linked_list=LinkedList()
        user_input= input("Enter the Linked List elements")
        elements=user_input.split()
        for element in elements:
            linked_list.append(int(element))
        return linked_list 

    if __name__=="__main__":
        linked_list=create_list()
        print("Original Linked List:")
        linked_list.print_list()

    index_to_remove=int(input("Enter the index to be removed:"))
    linked_list.remove_at_index(index_to_remove)

    print("Linked list after removing the element at the index",index_to_remove,":")
    linked_list.print_list()                                                   